docs/markdown/Using Pants/concepts/options.md

```diff
@@ -317,6 +317,35 @@ will set the value to `{'foo': 42, 'bar': 2, 'baz': 3}`, and
 
 will set the value to `{'foo': 42, 'baz': 3}`.
 
+Reading individual option values from files
+===========================================
+
+If an option value is too large or elaborate to use directly, or if you don't want to hard-code 
+values directly in `pants.toml`, you can set the value of any option to the string 
+`@relative/path/from/repo/root/to/file` (note the leading `@`), and the value will be read 
+from that file. 
+
+If the file name ends with `.json` or `.yaml` then the file will be parsed as the relevant
+format, which is useful for list- and dict-valued options. 
+
+Otherwise, the file is parsed as a literal as described above for each option type.
+
+Note that you can use this feature on the command-line, in an env var, or in a config file:
+
+```toml
```

Comment:
```suggestion
```toml pants.toml
```

Reply:
We set fromfile=True by default, and we never turn it off, so...

Reply:
Oh cool! Didn't realize that. Still, this little `pants.toml` change would be good + mentioning invalidation

Reply:
To be precise, we assume it's true if unset: https://github.com/pantsbuild/pants/blob/0d975f24129822700f581496ff66ac4368a2ef85/src/python/pants/option/parser.py#L559

We could probably remove the fromfile arg, I see no need to ever turn this off.